search/management/commands/post_dirty_sounds_to_solr.py

```diff
@@ -36,26 +36,16 @@ class Command(BaseCommand):
 
     def handle(self, *args, **options):
 
-        # Index all those which are processed and moderated ok
-        num_sounds = Sound.objects.filter(processing_state="OK", moderation_state="OK", is_index_dirty=True).count()
+        # Index all those which are processed and moderated ok that has is_index_dirty
+        sounds_to_index = Sound.objects.filter(processing_state="OK", moderation_state="OK", is_index_dirty=True)
+        num_sounds = sounds_to_index.count()
         logger.info("Starting posting dirty sounds to solr. %i sounds to be added/updated to the solr index"
                     % num_sounds)
 
-        num_correctly_indexed_sounds = 0
-        slice_size = 1000
-        for i in range(0, num_sounds, slice_size):
-            console_logger.info("Adding %i sounds to solr, slice %i", slice_size, i)
-            try:
-                # Get all sounds moderated and processed ok that has is_index_dirty
-                where = "sound.moderation_state = 'OK' AND sound.processing_state = 'OK' AND is_index_dirty = true " \
-                        "AND sound.id > %s"
-                order_by = "sound.id ASC"
-                sounds_qs = Sound.objects.bulk_query_solr(where, order_by, slice_size, (i, ))
-                add_sounds_to_solr(sounds_qs)
-                num_correctly_indexed_sounds += slice_size
-            except SolrException as e:
-                console_logger.error("failed to add sound batch to solr index, reason: %s", str(e))
-                raise
+        num_correctly_indexed_sounds = add_all_sounds_to_solr(sounds_to_index)
+
+        if num_correctly_indexed_sounds == num_sounds:
+            sounds_to_index.update(is_index_dirty=False)
```

Comment:
Why do we do this here? If a batch of sounds can't be indexed for some reason (see `try/except` in `add_all_sounds_to_solr` ), then it can happen that `num_correctly_indexed_sounds != num_sounds` and we wont mark them as index dirty. I think we should use the `mark_index_clean` option passed to `add_all_sounds_to_solr` that we had before (https://github.com/MTG/freesound/blob/f17064bf5d5419a3657dcd3dacf7d094d0b227d0/utils/search/search_general.py#L101-L103).

Reply:
The main reason of doing it in this way is to avoid doing one more query on each iteration. Also, I thought that if the indexing fails then it will fail for all the sounds so the condition was fine.